server/services/user.service.ts

```diff
@@ -60,43 +55,19 @@ export class UserService {
       throw new Error((err as Error).message);
     }
   }
-  generateTokens(userId: string): { token: string; refreshToken: string } {
-    const token = this.createNewToken(userId, "1d");
-    const refreshToken = this.createNewToken(userId, "3m");
-
-    return { token, refreshToken };
-  }
-  createNewToken(userId: string, expiresIn: "1d" | "3m"): string {
-    if (!appConfig.jwtSecret) {
-      throw Error("Invalid token secret");
-    }
-    return jwt.sign({ userId }, appConfig.jwtSecret, {
-      expiresIn,
-    });
-  }
-  verifyToken(token: string): jwt.JwtPayload | string | boolean {
-    if (!appConfig.jwtSecret) throw Error("Invalid token secret");
+  async refreshToken(token: string): Promise<string> {
     try {
-      return jwt.verify(token, appConfig.jwtSecret);
+      const { userId } = verifyToken(token) as IgenerateToken;
```

Comment:
why do we need to re-verify refreshToken? we are verifying this refreshToken in authToken middleware also, right?

Reply:
if you want userId from the token you can use `decode()` function from jsonwebtoken no need to verify token again

Reply:
Fixed